### ISSUE TITLE
add at least one enum to an example

### ISSUE THREAD

**Issue by chrisdrackett** [mobxjs/mst-gql#201]
Would be good for testing outside our tests.

**Comment by beepsoft** [comment]
I just tried 0.10.0 and actually there is something wrong with enum generation: both the model and the typescript type has the same name. Eg. for an enum `StatusEnum` there should be a file generated with name `StatusEnum.ts` with a typescript type called `StatusEnum` and an mst model `StatusEnumModel`, however the model is now also called `StatusEnum`.

I will look into this now.

**Comment by beepsoft** [comment]
@chrisdrackett It seems that you changed something around this in the a222426e964d8f646774720ec04a2b1e851e8027 and 11e79aa548281874f437eaff2b103da9dd174499 commits. My version generated the `...Enum` and `...EnumModel` names but now you generate the same name for both. Can you take a look at this?

**Comment by chrisdrackett** [comment]
Yeah, enums are not models, I had missed this during the review as none of the mst-gql examples have enums, but found it when updating our project. That being said, maybe I missed something? I don’t see any duplicate names in our enums, but there might be a case I’m not aware of.

**Comment by beepsoft** [comment]
Take a look at `tests/generator/generate.test.js`. There you test for this result: a typescript `enum InterestEnum` and an mst model/type `const InterestEnum`. Here we don't compile these with `tsc` but if we did it would fail because of the name clash.

```javascript
test("enums ending in Enum doesn't have an extra Enum postfix with namingConvention=js", () => {
 ...
  expect(interestEnumFile).toBeTruthy()
  expect(
    hasFileContentExact(interestEnumFile, "export enum InterestEnum {")
  ).toBeTruthy()
  expect(
    hasFileContentExact(
      interestEnumFile,
      'export const InterestEnum = types.enumeration("InterestEnum"'
    )
  ).toBeTruthy()
```

The simplest solution would be to give the mst type some new name. Maybe `InterestEnumType`?

**Comment by chrisdrackett** [comment]
ah, I see, we don't run into this in our app because our enums don't have `enum` in the name. So we end up with something like `export enum ItemType` and `export const ItemTypeEnum` in our `ItemTypeEnum` file. I think the suggestion of adding `Type` to the end of the type is probably the best thing to do here as it matches with what we do with models.

**Comment by beepsoft** [comment]
@chrisdrackett should I create a PR for fixing the above issue?

**Comment by chrisdrackett** [comment]
sure, sorry, we've been extremely busy so I haven't had much time to put to this.

**Comment by beepsoft** [comment]
I added PR #237 to fix this issue. Note that I ended up adding ...Model suffix to MST enum models instead adding the ...Type suffix to the TS enum type. This is because I just realized that enums were the only thing not having a Model suffix neither in their type names nor in their file names. So, this way all generated MST models are now name "SomeModel" everywhere.
